Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -139,7 +139,7 @@ public DataGrid()
         /// Links the child column with this datagrid.
         /// </summary>
         /// <param name="column">Column to link with this datagrid.</param>
-        internal void Hook( DataGridColumn<TItem> column )
+        public void AddColumn( DataGridColumn<TItem> column )

```

Comment:
Does this API actually work well, with adding columns programatically. It's different from when adding through a razor file which actually does some additional stuff behind the curtains.

For example :  `[CascadingParameter] protected DataGrid<TItem> ParentDataGrid { get; set; }` comes to mind, I think I had trouble implementing something similar to this before.

Reply:
Yeah, you're right. Maybe I can make cascaded ParentDataGrid public and then let developers assign it manually?

Reply:
Since it's a Parameter it should already be public. Maybe as it should be a part of DataGridColumn ctor... I don't think there's an easy way to go about this.

Reply:
Well unline Parameter, a CascadedParameter is supposed to be private. At least that is the convention. But nothing is set in stone so it might be a good idea to make it public. It is the DataGrid instance that is already public so it is good in this case.

Reply:
I was taking a look at the place where I had to handle something like that.
I personally used a **ctor**, which kinda makes it more noticeable and relevant, but I guess making the `CascadingParameter `public is fine also.
_xtabs => `CascadingParameter`
![image](https://user-images.githubusercontent.com/22283161/123551018-e34cda00-d767-11eb-80d3-51c933d1ee23.png)

If someone ends up using that method for some reason, if I'm not wrong there's going to be a runtime error due to CascadingParameter being null, and the developer will notice it and set the CascadingParameter manually.